src/components/Edit.js

```diff
@@ -188,10 +124,16 @@ class Edit extends Component {
                 </div>
               </div>
             </div>
-            <div>
-              {editorContainer}
-            </div>
-
+            <EditorArea
+              docName={name}
+              docType={type}
+              docKeys={rawKeys}
+              viewMode={viewMode}
+              onEditor={onEditor}
+              onEditorValueChange={onEditorValueChange}
+              snapshots={snapshots}
+              onTakeSnapshot={onTakeSnapshot}
+              previewMd={md} />
```

Comment:
What do you think about renaming `previewMd` and the `md` state? It looks like it's a cache of the editor contents whatever syntax that might be, and it happens to be used for previewing, but that feels like a implementation detail rather than a contract.

How about `editorText`? or `lastEditorText` if we wanna be extra explicit about the fact that it's a secondary source.